ddmrp/models/purchase_order.py

```diff
@@ -25,10 +25,25 @@ class PurchaseOrderLine(models.Model):
     ddmrp_comment = fields.Text(related="order_id.ddmrp_comment")
 
     def create(self, vals):
-        record = super(PurchaseOrderLine, self).create(vals)
+        record = super().create(vals)
         record._calc_execution_priority()
+        if record.product_id:
+            record._find_buffer_link()
         return record
 
+    def write(self, vals):
+        res = super().write(vals)
+        for rec in self:
+            if rec.product_id:
+                rec._find_buffer_link()
```

Comment:
instead of adding the same check on product_id everywhere, and calling `_find_buffer_link()`, why not calling `self._find_buffer_link()` and filter in this method at the same time as the filter on `not rec.buffer_ids`?